pkg/cmd/extension/command.go

```diff
@@ -49,6 +49,33 @@ func NewCmdExtension(f *cmdutil.Factory) *cobra.Command {
 		Aliases: []string{"extensions", "ext"},
 	}
 
+	upgradeFunc := func(name string, flagForce, flagDryRun bool) error {
+		cs := io.ColorScheme()
+		err := m.Upgrade(name, flagForce)
+		if err != nil {
+			if name != "" {
+				fmt.Fprintf(io.ErrOut, "%s Failed upgrading extension %s: %s\n", cs.FailureIcon(), name, err)
+			} else if errors.Is(err, noExtensionsInstalledError) {
+				return cmdutil.NewNoResultsError("no installed extensions found")
+			} else {
+				fmt.Fprintf(io.ErrOut, "%s Failed upgrading extensions\n", cs.FailureIcon())
+			}
+			return cmdutil.SilentError
+		}
+		if io.IsStdoutTTY() {
+			successStr := "Successfully"
+			if flagDryRun {
+				successStr = "Would have"
+			}
+			extensionStr := "extension"
+			if name == "" {
+				extensionStr = "extensions"
+			}
+			fmt.Fprintf(io.Out, "%s %s upgraded %s\n", cs.SuccessIcon(), successStr, extensionStr)
+		}
```

Comment:
This duplicates a significant portion of the `upgrade` command UI in a way that I don't think it's incidental duplication (it's essentially the same operation, so should be the same code path). Could the two see more reuse?

Reply:
Oops. This is an intentional copy of the `upgrade` command below but I forgot to replace that code with a call of this func. 🤦‍♂️